test/benchmark/bench_linux_and_aster.sh

```diff
@@ -80,39 +43,81 @@ parse_results() {
 # Run the benchmark on Linux and Asterinas
 run_benchmark() {
     local benchmark="$1"
-    local search_pattern="$2"
-    local result_index="$3"
+    local benchmark_type="$2"
+    local search_pattern="$3"
+    local result_index="$4"
 
     local linux_output="${BENCHMARK_DIR}/linux_output.txt"
     local aster_output="${BENCHMARK_DIR}/aster_output.txt"
     local result_template="${BENCHMARK_DIR}/${benchmark}/result_template.json"
     local benchmark_name=$(basename "${benchmark}")
+    local benchmark_root=$(dirname "${benchmark}")
     local result_file="result_${benchmark_name}.json"
     
     echo "Preparing libraries..."
     prepare_libs
 
-    local asterinas_cmd="make run BENCHMARK=${benchmark} ENABLE_KVM=1 RELEASE_LTO=1 2>&1 | tee ${aster_output}"
-    echo "Running benchmark ${benchmark} on Asterinas..."
-    eval "$asterinas_cmd"
-
-    prepare_fs
-    local linux_cmd="/usr/local/qemu/bin/qemu-system-x86_64 \
-        --no-reboot \
-        -smp 1 \
-        -m 8G \
-        -machine q35,kernel-irqchip=split \
-        -cpu Icelake-Server,-pcid,+x2apic \
-        --enable-kvm \
-        -kernel ${LINUX_KERNEL} \
-        -initrd ${BENCHMARK_DIR}/../build/initramfs.cpio.gz \
-        -drive if=none,format=raw,id=x0,file=${BENCHMARK_DIR}/../build/ext2.img \
-        -device virtio-blk-pci,bus=pcie.0,addr=0x6,drive=x0,serial=vext2,disable-legacy=on,disable-modern=off,queue-size=64,num-queues=1,config-wce=off,request-merging=off,write-cache=off,backend_defaults=off,discard=off,event_idx=off,indirect_desc=off,ioeventfd=off,queue_reset=off \
-        -append 'console=ttyS0 rdinit=/benchmark/common/bench_runner.sh ${benchmark} linux mitigations=off hugepages=0 transparent_hugepage=never' \
-        -nographic \
-        2>&1 | tee ${linux_output}"
-    echo "Running benchmark ${benchmark} on Linux..."
-    eval "$linux_cmd"
+    case "${benchmark_type}" in
+        "guest")
+            local asterinas_cmd="make run BENCHMARK=${benchmark} ENABLE_KVM=1 RELEASE_LTO=1 2>&1 | tee ${aster_output}"
+            local linux_cmd="/usr/local/qemu/bin/qemu-system-x86_64 \
```

Comment:
The asterinas_cmd and linux_cmd remain duplicates. How about the following change:

```shell

local asterinas_cmd="make run BENCHMARK=${benchmark} ENABLE_KVM=1 RELEASE_LTO=1 2>&1"
local linux_cmd="/usr/local/qemu/bin/qemu-system-x86_64 \
                --no-reboot \
                -smp 1 \
                -m 8G \
                -machine q35,kernel-irqchip=split \
                -cpu Icelake-Server,-pcid,+x2apic \
                --enable-kvm \
                -kernel ${LINUX_KERNEL} \
                -initrd ${BENCHMARK_DIR}/../build/initramfs.cpio.gz \
                -drive if=none,format=raw,id=x0,file=${BENCHMARK_DIR}/../build/ext2.img \
                -device virtio-blk-pci,bus=pcie.0,addr=0x6,drive=x0,serial=vext2,disable-legacy=on,disable-modern=off,queue-size=64,num-queues=1,config-wce=off,request-merging=off,write-cache=off,backend_defaults=off,discard=off,event_idx=off,indirect_desc=off,ioeventfd=off,queue_reset=off \
                -append 'console=ttyS0 rdinit=/benchmark/common/bench_runner.sh ${benchmark} linux mitigations=off hugepages=0 transparent_hugepage=never' \
                -netdev user,id=net01,hostfwd=tcp::5201-:5201 \
                -device virtio-net-pci,netdev=net01,disable-legacy=on,disable-modern=off \
                -nographic \
                2>&1"

case "${benchmark_type}" in
        "guest")
            asterinas_cmd=$asterinas_cmd" | tee ${aster_output}"
            linux_cmd=$linux_cmd" | tee ${linux_output}"
            echo "Running benchmark ${benchmark} on Asterinas..."
            eval "$asterinas_cmd"
            prepare_fs
            echo "Running benchmark ${benchmark} on Linux..."
            eval "$linux_cmd"
            ;;
        "host-guest")
            echo "Running benchmark ${benchmark} on host and guest..."
            bash "${BENCHMARK_DIR}/${benchmark_root}/${benchmark_root}_bench_runner.sh" \
                "${BENCHMARK_DIR}/${benchmark}" \
                "${asterinas_cmd}" \
                "${linux_cmd}" \
                "${aster_output}" \
                "${linux_output}"
            ;;
        "guest-guest")
            echo "Running benchmark ${benchmark} between guests..."
            echo "TODO"
            exit 1
            ;;
        *)
            echo "Error: Unknown benchmark type '${benchmark_type}'" >&2
            exit 1
            ;;
    esac

```

Reply:
I didn't realize the way of `" | tee ${linux_output}"`. Thank you for your suggestions!